src/scriptworker/cot/verify.py

```diff
@@ -1298,7 +1305,7 @@ async def get_in_tree_template(link):
 
     """
     context = link.context
-    source_url = get_source_url(link)
+    source_url = await get_source_url(context, link)
```

Comment:
I think it might make sense to split the behavior of ensuring that the URL points at `.taskcluster.yml` that we want here, from the rest of the behavior of `get_source_url`. 

It seems that we are relying on the fact that this call only happens when `link.task_type in PARENT_TASK_TYPES`. 

I *think* I would prefer if we used `get_repo` here, and constructed the appropriate URL (particularly given the fact that we deconstruct and reconstruct the URL here in some cases anyway).

The only niggiling is how much the validity of chain-of-trust hinges on `get_source_url` and `get_repo` matching up, and whether not calling `get_source_url` here changes that. I don't *think* so, as `trace_back_to_tree` calls `get_source_url` on everything in the chain already.

This also has the benefit (I think) of allowing the source to point to a pretty URL, rather than the raw one; all the non-decision tasks already do that, and I think the only reason the decision tasks don't is to satisfy scriptworker. If scriptworker doesn't need it, it provides a nicer UX for people looking at tasks.